packages/kiota_http/lib/src/http_client_request_adapter.dart

```diff
@@ -0,0 +1,272 @@
+part of '../kiota_http.dart';
+
+class HttpClientRequestAdapter implements RequestAdapter {
+  HttpClientRequestAdapter({
+    required http.Client client,
+    required AuthenticationProvider authProvider,
+    required ParseNodeFactory pNodeFactory,
+    required SerializationWriterFactory sWriterFactory,
+  })  : _client = client,
+        _authProvider = authProvider,
+        _pNodeFactory = pNodeFactory,
+        _sWriterFactory = sWriterFactory;
+
+  static const String _claimsKey = 'claims';
+
+  final http.Client _client;
+  final AuthenticationProvider _authProvider;
+  ParseNodeFactory _pNodeFactory;
+  SerializationWriterFactory _sWriterFactory;
+  String? _baseUrl;
+
+  @override
+  String? get baseUrl => _baseUrl;
+
+  @override
+  set baseUrl(String? value) {
+    if (value?.endsWith('/') ?? false) {
+      _baseUrl = value?.substring(0, value.length - 1);
+    } else {
+      _baseUrl = value;
+    }
+  }
+
+  Future<http.BaseRequest> _getMessageFromInfo(
+    RequestInformation requestInfo,
+  ) async {
+    final request = http.Request(requestInfo.httpMethod!.name, requestInfo.uri);
+
+    // TODO(ricardoboss): implement the rest of the method
+
+    return request;
+  }
+
+  Future<http.StreamedResponse> _getResponse(
+    RequestInformation requestInfo, [
+    String? claims,
+  ]) async {
+    _setBaseUrl(requestInfo);
+
+    Map<String, Object>? additionalAuthContext;
+    if (claims != null) {
+      additionalAuthContext = {_claimsKey: claims};
+    }
+
+    await _authProvider.authenticateRequest(requestInfo, additionalAuthContext);
+
+    final message = await _getMessageFromInfo(requestInfo);
+
+    return _client.send(message);
+  }
+
+  void _setBaseUrl(RequestInformation requestInfo) {
+    requestInfo.queryParameters['baseUrl'] = baseUrl;
+  }
+
+  ResponseHandler? _getResponseHandler(RequestInformation requestInfo) {
+    final option = requestInfo.getRequestOption<ResponseHandlerOption>();
+
+    return option?.responseHandler;
+  }
+
+  Future<void> _throwIfFailedResponse(
+    http.StreamedResponse response,
+    ErrorMappings? errorMapping,
+  ) async {
+    if (response.statusCode >= 200 && response.statusCode < 300) {
```

Comment:
This should probably change to <400 based on recent developments. Checkout https://github.com/microsoft/kiota-http-dotnet/issues/219 and related items.

Reply:
Should I only add an exception for `304`? Or special handling for `3xx` in general?

Reply:
For now, I guess its best to just not throw for `3xx`.